awpy/analytics/map_control.py

```diff
@@ -0,0 +1,382 @@
+"""Functions for calculating map control values and metrics.
+
+    Example notebook:
+
+    https://github.com/pnxenopoulos/awpy/blob/main/examples/05_Map_Control_Calculations_And_Visualizations.ipynb
+"""
+
+from collections import defaultdict
+
+import numpy as np
+
+from awpy.analytics.nav import area_distance, calculate_tile_area, find_closest_area
+from awpy.data import NAV, NAV_GRAPHS
+from awpy.types import FrameMapControl, GameFrame, GameRound, TeamFrameInfo
+
+
+def approx_neighbors(
+    map_name: str,
+    source_tile_id: int,
+    n_neighbors: int = 5,
+) -> list[tuple[float, int]]:
```

Comment:
Created a TileDistanceObject dataclass in new commit to address this issue.

Reply:
Noticed that my new TileDistanceObject is similar to the TileDistance object that already exists but it's not quite the same. I've noted down a TODO in types.py to merge these types together in the future - I don't believe it will be that hard to do. I believe the easiest way would be to introduce TileId fields for the 2 tiles in question for the TileDistance object. However, since the TileDistance object is used in other places, I think it's best to merge the two objects in a separate CR if that makes sense.